### PR TITLE
Feature/list no row divider

### PR DESCRIPTION
**Pull request by SilverGull**
- Row divider is optional - default is false to match the latest Kirby design for most lists.
- The @Input is named 'lines' to match Ionic
- As NativeScript does not support pseudo css selectors it will show a divider after the last row on Native. As it has the same color as the background (--kirby-background) you won't see it however (fixed when @askarby has implemented round corners)

### CODE REVIEW

File: src/kirby/components/list/list.component.scss [hunk 1]
```diff
@@ -17,19 +16,14 @@ ul {
...
       margin: 0;
     }
-    
-    
 
-    &.row{
-      &:first-of-type {
-        border-top: 1px solid var(--kirby-background);
-      }
+    &.divider:not(:last-child) {
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list.component.scss]: Could it not be a good idea to maybe call the class for `lines` also? 

Just to keep the same terminology.

File: src/kirby/components/list/list.component.tns.scss [hunk 2]
```diff
@@ -4,6 +4,10 @@
   padding: size('gutter') size('gutter-small') size('gutter') size('gutter-small');
   justify-content: space-between;
   flex-direction: row;
-  border-color: get-color('kirby-background');;
-  border-width: 0 0 1 0;
+
+  &.divider {
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list.component.tns.scss]: I can see you remove the line on the last element on web. Should this not also be done on native?

File: src/kirby/components/list/list.component.ts [hunk 3]
```diff
@@ -39,6 +39,7 @@ export class ListSectionHeaderDirective {}
 export class ListComponent implements OnInit {
   @Input() items: any[];
   @Input() getSectionName?: (item: any) => string;
+  @Input() lines = false;
```
Comment (jonasjensen-brolstark) [inline src/kirby/components/list/list.component.ts]: I can see you found inspiration in Ionic's [list](https://ionicframework.com/docs/api/list#properties) 👍

File: src/kirby/components/list/list.component.scss (hunk 1 above)
Comment (SilverGull) [reply src/kirby/components/list/list.component.scss]: I believe you are right - I just felt that "divider" was more saying :-) I have changed it.

File: src/kirby/components/list/list.component.tns.scss (hunk 2 above)
Comment (SilverGull) [reply src/kirby/components/list/list.component.tns.scss]: @askarby is working on rounded corners and adds a 'top' and 'bottom' class to rows. I can use that to remove the last line on native as well. I have added support for it, but it won't work until he lands his changes. Until then we will have a bottom line of one pixel the same color as the background, if lines are enabled.

File: src/kirby/components/list/list.component.ts (hunk 3 above)
Comment (SilverGull) [reply src/kirby/components/list/list.component.ts]: Yep :-)

**Comment by jonasjensen-brolstark** [comment]
LGTM

File: src/kirby/components/list/list.component.spec.ts [hunk 4]
```diff
@@ -71,6 +71,41 @@ describe('ListComponent', () => {
     });
   });
 
+  describe('lines', () => {
+    beforeEach(() => {
+      fixture = TestBed.createComponent(ListComponent);
+      component = fixture.componentInstance;
+    });
+
+    it('should set class "lines" on all li elements when lines is true', () => {
```
Comment (jakobe) [inline src/kirby/components/list/list.component.spec.ts]:
```suggestion
    it('should set class "divider" on all li elements when showDivider is true', () => {
```

File: src/kirby/components/list/list.component.spec.ts [hunk 5]
```diff
@@ -71,6 +71,41 @@ describe('ListComponent', () => {
...
+      component.items = [1, 2, 3];
+      component.lines = true;
+
+      fixture.detectChanges();
+
+      const rootElement: HTMLElement = fixture.debugElement.nativeElement;
+      const liElements = rootElement.querySelectorAll('li');
+
+      liElements.forEach((liElement) => {
+        expect(liElement.getAttribute('class')).toContain('lines');
```
Comment (jakobe) [inline src/kirby/components/list/list.component.spec.ts]:
```suggestion
        expect(liElement.getAttribute('class')).toContain('divider');
```

File: src/kirby/components/list/list.component.spec.ts [hunk 6]
```diff
@@ -71,6 +71,41 @@ describe('ListComponent', () => {
...
+      });
+    });
+
+    it('should not set class "lines" on any li elements when lines is false', () => {
```
Comment (jakobe) [inline src/kirby/components/list/list.component.spec.ts]:
```suggestion
    it('should not set class "divider" on any li elements when showDivider is false', () => {
```

File: src/kirby/components/list/list.component.spec.ts [hunk 7]
```diff
@@ -71,6 +71,41 @@ describe('ListComponent', () => {
...
+      component.items = [1, 2, 3];
+      component.lines = false;
+
+      fixture.detectChanges();
+
+      const rootElement: HTMLElement = fixture.debugElement.nativeElement;
+      const liElements = rootElement.querySelectorAll('li');
+
+      liElements.forEach((liElement) => {
+        expect(liElement.getAttribute('class')).not.toContain('lines');
```
Comment (jakobe) [inline src/kirby/components/list/list.component.spec.ts]:
```suggestion
        expect(liElement.getAttribute('class')).not.toContain('divider');
```

File: src/kirby/components/list/list.component.tns.html [hunk 8]
```diff
@@ -9,7 +9,7 @@
   >
     <!-- ITEM -->
     <ng-template tkListItemTemplate let-item="item">
-      <FlexboxLayout class="row" (tap)="onItemSelect(item)">
+      <FlexboxLayout class="row" [ngClass]="{'lines': lines}" (tap)="onItemSelect(item)">
```
Comment (jakobe) [inline src/kirby/components/list/list.component.tns.html]:
```suggestion
      <FlexboxLayout class="row" [ngClass]="{'divider': showDivider}" (tap)="onItemSelect(item)">
```

File: src/kirby/components/list/list.component.tns.scss [hunk 9]
```diff
@@ -4,8 +4,15 @@
   padding: size('gutter') size('gutter-small') size('gutter') size('gutter-small');
   justify-content: space-between;
   flex-direction: row;
-  border-color: get-color('kirby-background');;
-  border-width: 0 0 1 0;
+
+  &.lines {
```
Comment (jakobe) [inline src/kirby/components/list/list.component.tns.scss]:
```suggestion
  &.divider {
```

File: src/kirby/components/list/list.component.tns.scss [hunk 10]
```diff
@@ -4,8 +4,15 @@
...
+    border-color: get-color('kirby-background');
```
Comment (jakobe) [inline src/kirby/components/list/list.component.tns.scss]:
```suggestion
    border-color: get-color('kirby-grey-2');
```

File: src/kirby/components/list/list.component.tns.scss [hunk 11]
```diff
@@ -4,8 +4,15 @@
...
+    border-width: 0 0 1 0;
+    
+    &.bottom {
+      border-width: 0 0 0 0;
```
Comment (jakobe) [inline src/kirby/components/list/list.component.tns.scss]:
```suggestion
      border-width: 0;
```

File: src/kirby/components/list/list.component.tns.scss [hunk 12]
```diff
@@ -4,8 +4,15 @@
   padding: size('gutter') size('gutter-small') size('gutter') size('gutter-small');
   justify-content: space-between;
   flex-direction: row;
-  border-color: get-color('kirby-background');;
-  border-width: 0 0 1 0;
+
+  &.lines {
+    border-color: get-color('kirby-background');
+    border-width: 0 0 1 0;
```
Comment (jakobe) [inline src/kirby/components/list/list.component.tns.scss]:
```suggestion
    border-bottom-width: 1;
```

File: src/kirby/components/list/list.component.ts [hunk 13]
```diff
@@ -30,6 +30,7 @@ import { ListHelper } from './helpers/list-helper';
...
```
Comment (jakobe) [inline src/kirby/components/list/list.component.ts]:
```suggestion
  @Input() showDivider = false;
```

**Pull request merged**